core/src/main/java/jenkins/model/queue/AsynchronousExecution.java

```diff
@@ -106,13 +107,26 @@ public synchronized final Executor getExecutor() {
         return executor;
     }
 
+    /**
+     * Set the executor without notifying it about task completion.
+     * The caller <b>must</b> also call {@link #maybeComplete()}
+     * after releasing any problematic locks.
+     */
     @Restricted(NoExternalUse.class)
-    public synchronized final void setExecutor(@Nonnull Executor executor) {
```

Comment:
Javadoc errors—seems there were some forgotten references to this method in Javadoc.

Reply:
@jglick Good point -- fixed in latest commit.